### PR TITLE
Roll ANGLE, vulkan-deps, and shaderc

### PR DESCRIPTION
**Pull request by loic-sharma**
This change updates ANGLE to a version that contains [angle#3906325](https://chromium-review.googlesource.com/c/angle/angle/+/3906325), which fixes the COM error reported in  https://github.com/flutter/flutter/issues/110948. This also updates vulkan-deps and shaderc as these are dependencies of ANGLE.

Part of https://github.com/flutter/flutter/issues/110948

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]

File: DEPS [hunk 1]
```diff
@@ -109,7 +109,7 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + 'b7ef254d8cec53bdad39cdb4a284af1d0a79dbdb',
+  'src': 'https://github.com/loic-sharma/flutter-buildroot.git' + '@' + '3b88ba2edc4c7baf6d5bd6a48f270124edfef276',
```
Comment (loic-sharma) [inline DEPS:112]: ⚠️ TODO

File: DEPS [hunk 2]
```diff
@@ -109,7 +109,7 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + '64d07cbd3d9fa0c15f06c8e24c3bdbf5a9a06329',
```
Comment (loic-sharma) [inline DEPS:112]: TODO: https://github.com/flutter/engine/pull/36538#discussion_r984866234

File: DEPS (hunk 1 above)
Comment (loic-sharma) [reply DEPS]: VS 2019 was updated in a separate pull request, marking as resolved

**Comment by skia-gold** [comment]
Gold has detected about 16 new digest(s) on patchset 9.
View them at https://flutter-engine-gold.skia.org/cl/github/36389

File: tools/gn [hunk 3]
```diff
@@ -561,6 +566,7 @@ def to_gn_args(args):
                              get_host_os() == 'win'):
     # Do not build unnecessary parts of the ANGLE tree.
     gn_args['angle_build_all'] = False
+    gn_args['angle_has_astc_encoder'] = False
```
Comment (loic-sharma) [inline tools/gn:569]: ANGLE added a new dependency that we don't need. This opts-out of it.

**Review by dnfield — APPROVED** [review]

**Review by yaakovschectman — APPROVED** [review]

**Review by a-wallen — APPROVED** [review]

**Review by a-wallen — APPROVED** [review]

**Review by cbracken — APPROVED** [review]
lgtm

**Comment by Hixie** [comment]
test-exempt: rolling dependencies

**Pull request merged**